utils/src/foreign_field.rs

```diff
@@ -217,6 +220,10 @@ impl BigUintForeignFieldHelpers for BigUint {
         BigUint::two().pow(3 * LIMB_BITS as u32)
     }
 
+    fn max_foreign_field_modulus() -> Self {
+        BigUint::two().pow(259u32) - BigUint::one()
```

Comment:
Hmm..  Good point. I thought I computed it as exactly 259 bit max here in this section -> https://github.com/o1-labs/proof-systems/blob/master/book/src/rfcs/foreign_field_mul.md#parameter-selection

I better check it again!

Reply:
Completely nit, but I think it was `2^t * n ≥ f^2 - 1` but yeah, completely negligible. I mean I believe we said mmax to be just 259 as a "safe bound", even if we knew it was not very accurate. In any case It is not the number of bits of the foreign field modulus that we should care about, but the actual maximum modulus. Meaning not the power, but like the largest foreign field we could carry. And for that I guess you should find the largest prime that satisfies that bound... Which is a bit painful, but probably can be easily done with Sage.

Like, the question is: what's the largest prime `p` such that `2^264 * n ≥ p^2 - 1`

Reply:
Or even look them up here -> https://t5k.org/lists/2small/200bit.html

Reply:
Okay, checked with sage that the largest prime number of 259 bits (meaning this guy `2^259 - 361`) satisfies the bound `2^264 * n > p^2 - 1` for both `n = Pallas` and `n = Vesta`.

Reply:
That confirms that all foreign fields of 259 bits satisfy the bound. But since the logarithm is 259.5, one could find **some** 260-bit primes that also satisfy the bound. So let me find some...

Reply:
Amazing and very informative-- thanks!

Since must translate this calculation to code, specifically as a function of the native field modulus, which is inour case a rust trait and I don't think we want to implement or pay the runtime cost of the prime search, then do you think $m = \sqrt{t^2 \cdot n}$ is sufficient / close enough bound?

Reply:
Yes!

Reply:
Well, just for the sake of curiosity, the maximum foreign field we could support with `t=264` is the one with prime order `926336713898529563388567880069503262826888842373627227613104999999999999999607` which corresponds to `2^260 - 926336713898529563388567880069503262825430912276621797018216344126610074239369` (basically, 259.5 bits as we anticipated).

Reply:
better use `sqrt(2^t * n + 1)` in any case.

Reply:
Nice.  For comparison $floor(sqrt(2^t * n))$ for Pallas $n$ is `926336713898529563388567880069503262826888842373627227613105164261252000645135 > 926336713898529563388567880069503262826888842373627227613104999999999999999607`

Reply:
> Completely nit, but I think it was `2^t * n ≥ f^2 - 1` 

For completeness it works out to `2^t * n >= f^2` because it was derived (as in the RFC) from `2^t * n > f^2 - 1` which was derived from `a * b <= (f-1) * f + (f - 1) = f^2 - 1` but the approx won't ever be equal to this b/c of the sqrt and subsequent floor.

Reply:
Just for completeness, the hexadecimal value of the maximum foreign field is `80000000000000000000000000000000224698FC094CF91B9908B7C12F823FE77`